### PR TITLE
Bump promises dependency

### PR DESCRIPTION
**Pull request by ncooke3**

### CODE REVIEW

File: GoogleUtilities.podspec [hunk 1]
```diff
@@ -38,7 +38,7 @@ other Google CocoaPods. They're not intended for direct public usage.
   s.subspec 'Environment' do |es|
     es.source_files = 'GoogleUtilities/Environment/**/*.[mh]'
     es.public_header_files = 'GoogleUtilities/Environment/Public/GoogleUtilities/*.h'
-    es.dependency 'PromisesObjC', '~> 1.2'
+    es.dependency 'PromisesObjC', '>= 1.2', '<= 2.0'
```
Comment (paulb777) [inline GoogleUtilities.podspec:41]: Should be `< 3.0`, since we now want to allow any 2.x versions.

File: Package.swift [hunk 2]
```diff
@@ -63,7 +63,7 @@ let package = Package(
     ),
   ],
   dependencies: [
-    .package(name: "Promises", url: "https://github.com/google/promises.git", "1.2.8" ..< "1.3.0"),
+    .package(name: "Promises", url: "https://github.com/google/promises.git", "1.2.8" ... "2.0.0"),
```
Comment (paulb777) [inline Package.swift:66]: < "3.0.0"

**Review by paulb777 — APPROVED** [review]
Please confirm with a manual configure that Promises 2.0 gets installed both for SPM and pods. Then LGTM.

**Comment by ncooke3** [comment]
I've manually verified SPM & CocoaPods

**Pull request merged**